### PR TITLE
Fix QR send

### LINKED ISSUE #1999: When I scan a QR code, whether its a payment request or recipient public address, I want to be redirected with the right values filled in.
@omnat commented on [Mon Oct 19 2020](https://github.com/MetaMask/mobile-planning/issues/29)

Coinbase QR issue reported by support [here](https://consensys.slack.com/archives/GBW7S9FSN/p1606392113279800?thread_ts=1602872815.317800&cid=GBW7S9FSN).
They use another URL standard that doesn't work the same way as our payment request.

### PR DESCRIPTION
**Pull request by andrepimenta**
**Description**

PR that fixes QR code send flow:
1.  It goes to the right send flow from QR.
2. It shows a warning when the user inserts a token contract address or smart contract in the “to” field

### CODE REVIEW

File: app/core/DeeplinkManager.js [hunk 1]
```diff
@@ -136,22 +140,29 @@ class DeeplinkManager {
...
 				// Enforce https
+				onHandled && onHandled();
 				urlObj.set('protocol', 'https:');
 				this.handleBrowserUrl(urlObj.href, browserCallBack);
 				break;
 
 			// Specific to the MetaMask app
 			// For ex. go to settings
 			case 'metamask':
+				onHandled && onHandled();
```
Comment (rickycodes) [inline app/core/DeeplinkManager.js:159]: maybe we could consolodate these into a single method? eg:

```
const handle = () => onHandled && onHandled();
```
and then just call that?

**Comment by ibrahimtaveras00** [comment]
All scenarios look good to me 👍 

There's just 2 instances where I noticed we don't provide any feedback and just keep you on the send to screen:
1) when you're on the send to view and you scan a private key QR code
2) when you're on the send to view and you scan a seed phrase QR code

Do we want to do anything in this regard @andrepimenta @omnat @cjeria 

This is what we currently do if a user is on the wallet screen and they scan such things:
- Private Key

Screen Shot 2020-10-29 at 7 59 00 PM

-  Seed phrase

Screen Shot 2020-10-29 at 7 59 15 PM

Perhaps we send them to wallet with this messaging and continue from there?

**Review by ibrahimtaveras00 — APPROVED** [review]

File: locales/en.json [hunk 2]
```diff
@@ -580,7 +580,10 @@
 		"pending": "Pending",
 		"submitted": "Submitted",
 		"failed": "Failed",
-		"cancelled": "Cancelled"
+		"cancelled": "Cancelled",
+		"tokenContractAddressWarning": "WARNING: This address is a token contract address. If you send tokens to this address, you will lose them.",
+		"smartContractAddressWarning": "WARNING: This address is a smart contract address. Please make sure you understand what this address is for, otherwise you risk losing your funds.",
+		"continueError": "I understand the risks, continue"
```
Comment (rickycodes) [inline locales/en.json:586]: `es.json`?

**Review by rickycodes — CHANGES_REQUESTED** [review]

**Comment by omnat** [comment]
@andrepimenta @cjeria based off of our discussion in team standup, could we remove the warning when user enters just a contract address (which isn't a token address)?
It's totally my bad for not considering the case where people are actually intending to send tokens to smart contract accounts. I agree with Dan that the user error/mistakes are most of the times when sending to token addresses.

File: locales/en.json (hunk 2 above)
Comment (andrepimenta) [reply locales/en.json:586]: Added!

File: app/core/DeeplinkManager.js [hunk 3]
```diff
@@ -78,12 +78,15 @@ class DeeplinkManager {
 			}
 		}
 
+		const handled = () => onHandled?.();
```
Comment (rickycodes) [inline app/core/DeeplinkManager.js:81]: :heart_eyes:

**Review by rickycodes — APPROVED** [review]
:+1:

**Comment by andrepimenta** [comment]
> @andrepimenta @cjeria based off of our discussion in team standup, could we remove the warning when user enters just a contract address (which isn't a token address)?
> It's totally my bad for not considering the case where people are actually intending to send tokens to smart contract accounts. I agree with Dan that the user error/mistakes are most of the times when sending to token addresses.

Done it will show a warning:
Simulator Screen Shot - iPhone 11 Pro - 2020-11-04 at 20 32 15

**Review by ibrahimtaveras00 — APPROVED** [review]
Update looks good on both OS's 👍

**Review by ibrahimtaveras00 — APPROVED** [review]
Latest fixes look good 👍

**Pull request merged**